### PR TITLE
chore(Cross): [IOAPPX-270] Update `io-app-design-system` to `1.31.0`

### PR DESCRIPTION
**Pull request by dmnplb**
## Short description
This PR updates the `io-app-design-system` to the latest version available.

## List of changes proposed in this pull request
- Update `io-app-design-system` to `1.31.0`

## How to test
N/A

### CODE REVIEW

**Comment by shadowsheep1** [comment]
> Snapshots update needed.

@dmnplb why this changes?

image

File: ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap [hunk 1]
```diff
@@ -510,7 +517,7 @@ exports[`Test QtspClauseListItem component should render a QtspClauseListItem co
                               }
                             >
                               <View
-                                accessible={true}
+                                accessible={false}
```
Comment (shadowsheep1) [inline ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: Couldn't be this problematic in term of a11y? Why we have this change in termo of a11y?

**Review by shadowsheep1 — CHANGES_REQUESTED** [review]

File: ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap (hunk 1 above)
Comment (dmnplb) [reply ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: @shadowsheep1 This is the result of the PR relative to the `eslint-plugin-react-native-a11y`:
* https://github.com/pagopa/io-app-design-system/pull/227

Some `Selection` components, like `AnimatedCheckbox` or `AnimatedRadio` can't live alone (just the box/circle with the tick, nothing else), but they should be included in other components (like `CheckboxLabel` or `ListItemCheckbox`). For this reason, I preferred to set `accessible={false}` on those atomic components.

Comment (dmnplb) [reply ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: This snapshot could give us an indication of a wrong approach to building the component (for example, a custom, non-standard component that is not mapped to the DS).

Comment (shadowsheep1) [reply ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: Okay then. So this will improve a11y.

**Review by shadowsheep1 — APPROVED** [review]
LGTM

File: ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap (hunk 1 above)
Comment (shadowsheep1) [reply ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: > This snapshot could give us an indication of a wrong approach to building the component (for example, a custom, non-standard component that is not mapped to the DS).

@dmnplb  Wait, are you telling this should be check by the component owners? Maybe we should tag them?

Comment (dmnplb) [reply ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: @shadowsheep1 Just checked. The `SignatureFieldItem` correctly uses `ListItemCheckbox` (which manages the accessibility state at a parent level), so there's no accessibility regression. @LazyAfternoons Can you confirm?

Comment (shadowsheep1) [reply ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: @hevelius @LazyAfternoons have a look if after the merge of this PR you should address something for a11y in those components.

Comment (LazyAfternoons) [reply ts/features/fci/components/__tests__/__snapshots__/QtspClauseListItem.test.tsx.snap:520]: No regression should arise but we'll check, thank you.

**Pull request merged**